Xamarin.Forms.Platform.Android/AppCompat/SwitchRenderer.cs

```diff
@@ -81,6 +87,36 @@ protected override void OnElementChanged(ElementChangedEventArgs<Switch> e)
 
 				e.NewElement.Toggled += HandleToggled;
 				Control.Checked = e.NewElement.IsToggled;
+				defaultOnColor = Control.TrackTintList;
+				defaultMode = Control.TrackTintMode;
+				UpdateOnColor();
+			}
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == Switch.OnColorProperty.PropertyName)
+				UpdateOnColor();
+		}
+
+		private void UpdateOnColor()
+		{
+			if (Element != null)
+			{
+				if (Element.OnColor == Color.Default)
+				{
+					Control.TrackTintList = defaultOnColor;
+					Control.TrackTintMode = defaultMode;
+				}
+				else
+				{
+					if (Control.Checked)
+						Control.TrackTintList = ColorStateList.ValueOf(Element.OnColor.ToAndroid());
+					else
+					{
+						Control.TrackTintList = defaultOnColor;
+					}
```

Comment:
I'm afraid this implementation messes up the default behaviour. A switch where you don't set any color will get a different colors both in on and off mode compared to how it works without this change. Could be that you need to make sure the color is only set for the `global::Android.Resource.Attribute.StateChecked` state?

Reply:
I tried with that but it was showing square colored box instead of rounded corners

Reply:
Hello,

In my custom renderer for have the good visual, I just do like that ==>

      private void OnCheckedChange(object sender, CompoundButton.CheckedChangeEventArgs e)
        {
            if (this.Control.Checked)
            {
                this.Control.ThumbDrawable.SetColorFilter(_greenColor, PorterDuff.Mode.SrcAtop);
                this.Control.TrackDrawable.SetColorFilter(_greenColor, PorterDuff.Mode.Multiply);
            }
            else
            {
                this.Control.ThumbDrawable.SetColorFilter(_greyColor, PorterDuff.Mode.SrcAtop);
                this.Control.TrackDrawable.SetColorFilter(_greyColor, PorterDuff.Mode.Multiply);
            }
        }

Hopes that help

Reply:
@jgiacomini  I used that in first commit but I was told to use Control.TrackTintList instead.

Reply:
If TrackTintList doesn't work right then obviously lets go back tot he original approach.

Reply:
@jassmith okay I will revert it back to my last push that was working fine.

Reply:
@ravinderjangra just make sure it works for "all" Android versions (especially newest one). When I tried last time I think it didn't work everywhere. I'm happy to help try it out later this week or beginning next.

Reply:
@henricm  please suggest how should I implement this or should I revert back to last commit. In that it was working find but with filters.

Reply:
@ravinderjangra I'm afraid I have no fix ready, but I'm happy to try out once you push something that works. 
Please try to verify on as many Android versions as possible and also make sure the default behaviour of the Switch is intact. Maybe test setting some different default styles in your Android xml:s and see that this works as expected even with your fix - both in the case of when setting the `OnColor` and when left untouched.